src/pip/_internal/build_env.py

```diff
@@ -18,6 +19,25 @@
 logger = logging.getLogger(__name__)
 
 
+class _Prefix:
```

Comment:
This is the change from @benoit-pierre to allow installing into the build env twice. I don't honestly understand the logic here, so I'll defer to him - if he wants to add a docstring, I'll do so (just let me know what it should say!) otherwise I'll leave this.